Makefile

```diff
@@ -30,6 +35,12 @@ LD_FLAGS += -X github.com/k0sproject/k0s/pkg/build.EtcdVersion=$(etcd_version)
 LD_FLAGS += -X github.com/k0sproject/k0s/pkg/build.KonnectivityVersion=$(konnectivity_version)
 LD_FLAGS += -X \"github.com/k0sproject/k0s/pkg/build.EulaNotice=$(EULA_NOTICE)\"
 LD_FLAGS += -X github.com/k0sproject/k0s/pkg/telemetry.segmentToken=$(SEGMENT_TOKEN)
+LD_FLAGS += -X k8s.io/component-base/version.gitVersion="v$(KUBECTL_VERSION)"
```

Comment:
If these are only used for the kubectl version, maybe the variable names should be something like `kubectlGitVersion`?

Reply:
The constants are coming from 3rd party package, unfortunately, so we can't control naming and the env variables already have KUBECTL_ prefix (apart from the BUILDDATE which in my opinion generic one)